src/topology/sheaves/skyscraper.lean

```diff
@@ -235,4 +235,152 @@ def skyscraper_sheaf_functor [has_products.{u} C] : C ⥤ sheaf C X :=
+      category.id_comp, comp_eq_to_hom_iff, category.assoc, eq_to_hom_trans, eq_to_hom_refl,
+      category.comp_id, category_theory.functor.id_map],
+  end }
+
+end stalk_skyscraper_presheaf_adjunction_auxs
+
+section
+
+open stalk_skyscraper_presheaf_adjunction_auxs
+
+/--
+`skyscraper_presheaf_functor` is the right adjoint of `presheaf.stalk_functor`
+-/
+def skyscraper_presheaf_stalk_adjunction [has_colimits C] :
+  (presheaf.stalk_functor C p₀ : presheaf C X ⥤ C) ⊣ skyscraper_presheaf_functor p₀ :=
+{ hom_equiv := λ c 𝓕,
+  { to_fun := to_skyscraper_presheaf _,
+    inv_fun := from_stalk _,
+    left_inv := from_stalk_to_skyscraper _,
+    right_inv := to_skyscraper_from_stalk _ },
+  unit := stalk_skyscraper_presheaf_adjunction_auxs.unit _,
+  counit := stalk_skyscraper_presheaf_adjunction_auxs.counit _,
+  hom_equiv_unit' := λ 𝓕 c α,
+  begin
+    ext U, simp only [equiv.coe_fn_mk, to_skyscraper_presheaf_app, nat_trans.comp_app,
+      skyscraper_presheaf_functor.map'_app, skyscraper_presheaf_functor_map, unit_app], split_ifs,
+    { erw [category.id_comp, ←category.assoc, comp_eq_to_hom_iff, category.assoc, category.assoc,
+        category.assoc,  category.assoc, eq_to_hom_trans, eq_to_hom_refl, category.comp_id,
+        ←category.assoc _ _ α, eq_to_hom_trans, eq_to_hom_refl, category.id_comp], },
+    { apply ((if_neg h).symm.rec terminal_is_terminal).hom_ext }
+  end,
+  hom_equiv_counit' := λ 𝓕 c α,
+  begin
+    ext U, simp only [equiv.coe_fn_symm_mk, counit_app],
+    erw [colimit.ι_desc, ←category.assoc, colimit.ι_map, whisker_left_app, category.assoc,
+      colimit.ι_desc], refl,
+  end }
+
+instance [has_colimits C] : is_right_adjoint (skyscraper_presheaf_functor p₀ : C ⥤ presheaf C X) :=
+⟨_, skyscraper_presheaf_stalk_adjunction _⟩
+
+instance [has_colimits C] : is_left_adjoint (presheaf.stalk_functor C p₀) :=
+⟨_, skyscraper_presheaf_stalk_adjunction _⟩
+
+/--
+Taking stalks of a sheaf is the left adjoint functor to `skyscraper_sheaf_functor`
+-/
+def stalk_skyscraper_sheaf_adjunction [has_colimits C] [has_products.{u} C] :
+  sheaf.forget C X ⋙ presheaf.stalk_functor _ p₀ ⊣ skyscraper_sheaf_functor p₀ :=
+{ hom_equiv := λ 𝓕 c,
+  ⟨λ f, ⟨to_skyscraper_presheaf p₀ f⟩, λ g, from_stalk p₀ g.1, from_stalk_to_skyscraper p₀,
+   λ g, by { ext1, apply to_skyscraper_from_stalk }⟩,
+  unit :=
+  { app := λ 𝓕, ⟨(stalk_skyscraper_presheaf_adjunction_auxs.unit p₀).app 𝓕.1⟩,
+    naturality' := λ 𝓐 𝓑 ⟨f⟩,
+      by { ext1, apply (stalk_skyscraper_presheaf_adjunction_auxs.unit p₀).naturality } },
+  counit := stalk_skyscraper_presheaf_adjunction_auxs.counit p₀,
+  hom_equiv_unit' := λ 𝓐 c f,
+    by { ext1, exact (skyscraper_presheaf_stalk_adjunction p₀).hom_equiv_unit },
+  hom_equiv_counit' := λ 𝓐 c f, (skyscraper_presheaf_stalk_adjunction p₀).hom_equiv_counit }
+
+instance [has_colimits C] [has_products.{u} C] :
+  is_right_adjoint (skyscraper_sheaf_functor p₀ : C ⥤ sheaf C X) :=
+⟨_, stalk_skyscraper_sheaf_adjunction _⟩
```

Comment:
Sorry for this, but I also added adjunction of `skyscraper_sheaf` version, can you have a look again @erdOne